tests/api/test_datasets.py

```diff
@@ -1,91 +1,79 @@
     assert response.status_code == status.HTTP_200_OK
-
-    retrieved_dataset = response.json()
-    assert retrieved_dataset["name"] == dataset.name
-    assert retrieved_dataset["id"] == dataset.id
+    assert_response(response.json(), dataset)
 
 
-def test_create_dataset(db: Session, client: TestClient, normal_user_headers: dict[str, str]):
+def test_create_dataset(db: Session, client: TestClient, user_header: dict[str, str]):
     """
     Test creating a dataset.
     """
-    create_request = dataset_create_request(db, normal_user_headers)
+    create_request = dataset_create_request()
 
-    response = client.post("/datasets/", headers=normal_user_headers, data=create_request.json())
+    response = client.post("/datasets/", headers=user_header, content=create_request.model_dump_json())
     assert response.status_code == status.HTTP_200_OK
-
-    created_dataset = response.json()
-    assert created_dataset["name"] == create_request.name
-    assert created_dataset["description"] == create_request.description
+    assert_response(response.json(), create_request)
 
 
-def test_create_existing_dataset(db: Session, client: TestClient, normal_user_headers: dict[str, str]):
+def test_create_existing_dataset(db: Session, client: TestClient, user_header: dict[str, str]):
     """
     Test creating an existing dataset.
     """
-    existing_dataset = dataset_create(db, normal_user_headers)
-    create_request = DatasetCreate(**jsonable_encoder(existing_dataset))
-    response = client.post("/datasets/", headers=normal_user_headers, data=create_request.json())
+    create_request = dataset_create_request()
+    response = client.post("/datasets/", headers=user_header, content=create_request.model_dump_json())
+    assert response.status_code == status.HTTP_200_OK
+    response = client.post("/datasets/", headers=user_header, content=create_request.model_dump_json())
     assert response.status_code == status.HTTP_409_CONFLICT
 
 
-def test_update_dataset(db: Session, client: TestClient, normal_user_headers: dict[str, str]):
+def test_update_dataset(db: Session, client: TestClient, user_header: dict[str, str]):
     """
     Test updating a dataset.
     """
-    existing_dataset = dataset_create(db, normal_user_headers)
+    existing_dataset = new_dataset(db, user_header, hours_per_time_step=2)
 
-    update_request = DatasetUpdate(**jsonable_encoder(existing_dataset))
-    update_request.name = f"New Dataset Name-{random_lower_string()}"
-    update_request.description = f"New Dataset Description-{random_lower_string()}"
+    update_request = DatasetUpdate(
+        name=f"New Dataset Name-{random_string()}",
+        description=f"New Dataset Description-{random_string()}",
+    )
 
-    response = client.put(f"/datasets/{existing_dataset.id}", headers=normal_user_headers, data=update_request.json())
+    response = client.put(f"/datasets/{existing_dataset.id}", headers=user_header, content=update_request.model_dump_json())
```

Comment:
```suggestion
    response = client.put(f"/datasets/{existing_dataset.id}", headers=user_header, json=update_request.model_dump_json())
```
same as here and everywhere..?
Maybe that does not work for whatever reason?

Reply:
It's works if .model_dump() (returns a dict) is passed in instead. .model_dump_json() returns a raw json string. I changed it from data= to content= because using data= with text content raises a deprecation warning. Is it better to test with json as dict or as raw strings? I think both are equivalent

Reply:
both using content vs json results in the same request, except:
* using json will probably set the content-type correctly
* using json lets fastapi handle the serialization of the dict

So I would always prefer to use the `json` param when sendin json through fastapi or requests